homeassistant/components/powerwall/__init__.py

```diff
@@ -222,30 +222,30 @@ async def _login_and_fetch_base_info(
 async def _call_base_info(power_wall: Powerwall, host: str) -> PowerwallBaseInfo:
     """Return PowerwallBaseInfo for the device."""
 
-    (
-        gateway_din,
-        site_info,
-        status,
-        device_type,
-        serial_numbers,
-    ) = await asyncio.gather(
-        power_wall.get_gateway_din(),
-        power_wall.get_site_info(),
-        power_wall.get_status(),
-        power_wall.get_device_type(),
-        power_wall.get_serial_numbers(),
-    )
-
-    # Serial numbers MUST be sorted to ensure the unique_id is always the same
-    # for backwards compatibility.
-    return PowerwallBaseInfo(
-        gateway_din=gateway_din.upper(),
-        site_info=site_info,
-        status=status,
-        device_type=device_type,
-        serial_numbers=sorted(serial_numbers),
-        url=f"https://{host}",
-    )
+    try:
+        async with asyncio.TaskGroup() as tg:
+            gateway_din = tg.create_task(power_wall.get_gateway_din())
+            site_info = tg.create_task(power_wall.get_site_info())
+            status = tg.create_task(power_wall.get_status())
+            device_type = tg.create_task(power_wall.get_device_type())
+            serial_numbers = tg.create_task(power_wall.get_serial_numbers())
+            batteries = tg.create_task(power_wall.get_batteries())
+
+        # Serial numbers MUST be sorted to ensure the unique_id is always the same
+        # for backwards compatibility.
+        return PowerwallBaseInfo(
+            gateway_din=gateway_din.result().upper(),
+            site_info=site_info.result(),
+            status=status.result(),
+            device_type=device_type.result(),
+            serial_numbers=sorted(serial_numbers.result()),
+            url=f"https://{host}",
+            batteries=batteries.result(),
+        )
+    except ExceptionGroup as e:
+        raise e.exceptions[0] from None
+    except BaseExceptionGroup as e:
+        raise e.exceptions[0] from None
 
```

Comment:
https://developers.home-assistant.io/docs/creating_component_code_review/#5-make-your-pull-request-as-small-as-possible

Please do the task group refactor in a separate PR